lighthouse-core/lib/sd-validation/sd-validation.js

```diff
@@ -64,12 +62,73 @@ module.exports = async function validate(textInput) {
   if (schemaOrgErrors.length) {
     return schemaOrgErrors.map(error => {
       return {
-        validator: /** @type {ValidatorType} */ ('schema-org'),
+        validator: /** @type {LH.StructuredData.ValidatorType} */ ('schema-org'),
         path: error.path,
         message: error.message,
+        lineNumber: error.path ? getLineNumberFromJsonLDPath(inputObject, error.path) : null,
+        invalidTypes: error.invalidTypes,
       };
     });
   }
 
   return [];
 };
+
+/**
+ * @param {*} obj
+ * @param {string} path
+ * @returns null | number - line number of the path value in the prettified JSON
+ */
+function getLineNumberFromJsonLDPath(obj, path) {
+  // To avoid having an extra dependency on a JSON parser we set a unique key in the
+  // object and then use that to identify the correct line
+  const searchKey = Math.random().toString();
+  obj = JSON.parse(JSON.stringify(obj));
+
+  setValueAtJsonLDPath(obj, path, searchKey);
+  const jsonLines = JSON.stringify(obj, null, 2).split('\n');
+  const lineIndex = jsonLines.findIndex(line => line.includes(searchKey));
+
+  return lineIndex === -1 ? null : lineIndex + 1;
+}
+
+/**
+ * @param {*} obj
+ * @param {string} path
+ * @param {*} value
+ */
+function setValueAtJsonLDPath(obj, path, value) {
+  const pathParts = path.split('/').filter(p => !!p);
+  let currentObj = obj;
+  pathParts.forEach((pathPart, i) => {
+    if (pathPart === '0' && !Array.isArray(currentObj)) {
+      // jsonld expansion turns single values into arrays
+      return;
+    }
+
+    const isLastPart = pathParts.length - 1 === i;
+    let keyFound = false;
+    for (const key of Object.keys(currentObj)) {
+      // The actual key in JSON might be an absolute IRI like "http://schema.org/author"
+      // but key provided by validator is "author"
+      const keyParts = key.split('/');
+      const relativeKey = keyParts[keyParts.length - 1];
+      if (relativeKey === pathPart && currentObj[key] !== undefined) {
```

Comment:
seems like if we have `currentObj[key]` as `undefined` we could just return before this whole loop

Reply:
True, I don't see why we'd need it at all actually...